[Bug]: `pandas v2.1` breaks tests

### The Issue

`pandas` [recently released](https://pandas.pydata.org/docs/whatsnew/v2.1.0.html) `v2.1.0` which now breaks 12 tests. The pandas version has been locked to be `<2.1` in Pr #193.  

### Expected Behavior

All tests should pass

### Steps To Reproduce

1. Remove the `pandas<2.1` restriction in the `docs/requirements.txt` and `setup.cfg` file 
2. Run `tox -r`

### Log output

```shell
============================================================ short test summary info ============================================================
FAILED tests/test_input.py::TestExpandDefaults::test_expand_parameters_defaults[multi_index_empty] - AssertionError: MultiIndex level [2] are different
FAILED tests/test_read_strategies.py::TestReadGlpk::test_read_model - pandas.errors.ParserError: Error tokenizing data. C error: Expected 6 fields in line 7, saw 7
FAILED tests/test_read_strategies.py::TestReadGlpk::test_read_solution - pandas.errors.ParserError: Error tokenizing data. C error: Expected 6 fields in line 7, saw 7
FAILED tests/test_read_strategies.py::TestReadGlpk::test_merge_model_sol - pandas.errors.ParserError: Error tokenizing data. C error: Expected 6 fields in line 7, saw 7
FAILED tests/test_read_strategies.py::TestReadGlpk::test_convert_to_dataframe - pandas.errors.ParserError: Error tokenizing data. C error: Expected 6 fields in line 7, saw 7
FAILED tests/test_read_strategies.py::TestReadGlpk::test_convert_to_dataframe_error - pandas.errors.ParserError: Error tokenizing data. C error: Expected 6 fields in line 7, saw 7
FAILED tests/results/test_results_package.py::TestAccumulatedNewCapacity::test_individual - AssertionError: Length of new_levels (2) must be same as self.nlevels (3)
FAILED tests/results/test_results_package.py::TestAccumulatedNewCapacity::test_overlapping - AssertionError: Length of new_levels (2) must be same as self.nlevels (3)
FAILED tests/results/test_results_package.py::TestAccumulatedNewCapacity::test_overlapping_bitty - AssertionError: Length of new_levels (2) must be same as self.nlevels (3)
FAILED tests/results/test_results_package.py::TestDiscountFactorStorage::test_dfs_start - AssertionError: Length of new_levels (2) must be same as self.nlevels (3)
FAILED tests/results/test_results_package.py::TestDiscountFactorStorage::test_dfs_mid - AssertionError: Length of new_levels (2) must be same as self.nlevels (3)
FAILED tests/results/test_results_package.py::TestDiscountFactorStorage::test_dfs_end - AssertionError: Length of new_levels (2) must be same as self.nlevels (3)
```


### Operating System

Linux

### What version of otoole are you running?

1.1.0

### Possible Solution

_No response_

### Anything else?

_No response_

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -2,6 +2,10 @@
 Changelog
 =========
 
+(Development) Version 1.1.3
+===========================
+- Lock pandas to 2.1.4 or later
+
 Version 1.1.2
 =============
 - Update zenodo metadata for JOSS

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -4,7 +4,7 @@ flatten_dict
 Jinja2<3.1
 networkx
 openpyxl
-pandas>=1.1,<2.1
+pandas>=2.1.4
 pydantic>=2
 pydot
 pyyaml

--- a/setup.cfg
+++ b/setup.cfg
@@ -47,7 +47,7 @@ install_requires =
     xlrd
     pyyaml
     pydot
-    pandas>=1.1,<2.1
+    pandas>=2.1.4
     Amply>=0.1.6
     networkx
     flatten_dict

--- a/src/otoole/input.py
+++ b/src/otoole/input.py
@@ -341,8 +341,11 @@ class WriteStrategy(Strategy):
             df_default["VALUE"] = default_values[name]
 
             # combine result and default value dataframe
-            df = pd.concat([data, df_default])
-            df = df[~df.index.duplicated(keep="first")]
+            if not data.empty:
+                df = pd.concat([data, df_default])
+                df = df[~df.index.duplicated(keep="first")]
+            else:
+                df = df_default
             df = df.sort_index()
             output[name] = df
 

--- a/tests/test_input.py
+++ b/tests/test_input.py
@@ -265,7 +265,7 @@ class TestExpandDefaults:
         input_data_single_index_empty(region),
     ]
     parameter_test_data_ids = [
-        "multi_index_no_defaluts",
+        "multi_index_no_defaults",
         "multi_index",
         "multi_index_empty",
         "single_index",

